components/brave_shields/browser/ad_block_service.cc

```diff
@@ -123,21 +81,22 @@ void AdBlockService::SourceProviderObserver::OnDATLoaded(
 
 void AdBlockService::SourceProviderObserver::OnResourcesLoaded(
     const std::string& resources_json) {
-  if (dat_buf_.empty()) {
+  if (!filter_set_) {
     task_runner_->PostTask(
         FROM_HERE,
         base::BindOnce(&AdBlockEngine::UseResources,
                        adblock_engine_->AsWeakPtr(), resources_json));
   } else {
     auto engine_load_callback = base::BindOnce(
-        [](base::WeakPtr<AdBlockEngine> engine, bool deserialize,
-           DATFileDataBuffer dat_buf, const std::string& resources_json) {
+        [](base::WeakPtr<AdBlockEngine> engine,
+           rust::Box<adblock::FilterSet> filter_set,
+           const std::string& resources_json) {
           if (engine) {
-            engine->Load(deserialize, std::move(dat_buf), resources_json);
+            engine->Load(std::move(filter_set), resources_json);
           }
         },
-        adblock_engine_->AsWeakPtr(), deserialize_, std::move(dat_buf_),
-        resources_json);
+        adblock_engine_->AsWeakPtr(), *std::move(filter_set_), resources_json);
```

Comment:
I think you might have something a bit weird here with `*std::move(filter_set_)`. You want to move the underlying `rust::Box<adblock::FilterSet>`, not the `absl::optional` so I think it should be `std::move(*filter_set_)`